extensions/fbc/src/org/sbml/jsbml/ext/fbc/converters/FbcV1ToCobraConverter.java

```diff
@@ -53,13 +55,129 @@
+    }
+    
+    LocalParameter upperBoundParameter = kineticLaw.getLocalParameter(CobraConstants.UPPER_BOUND);
+    if (upperBoundParameter == null) 
+    {
+      // 3. Assign lower and upper flux bounds if they were set, use defaults otherwise
+      upperBoundParameter = reaction.getKineticLaw().createLocalParameter(CobraConstants.UPPER_BOUND);
+      if (this.defaultUpperBound != null)
+      {
+        upperBoundParameter.setValue(this.defaultUpperBound);
+      } else 
+      {
+        upperBoundParameter.setValue(Double.POSITIVE_INFINITY);
+      }
+      upperBoundParameter.setExplicitlyConstant(true);
+      upperBoundParameter.setUnits(CobraConstants.mmol_per_gDW_per_hr);
+    }
+    
+    LocalParameter lowerBoundParameter = kineticLaw.getLocalParameter(CobraConstants.LOWER_BOUND);
+    if (lowerBoundParameter == null) 
+    {
+      lowerBoundParameter = kineticLaw.createLocalParameter(CobraConstants.LOWER_BOUND);
+      if (this.defaultLowerBound != null)
+      {
+        lowerBoundParameter.setValue(this.defaultLowerBound);  
+      } else 
+      {
+        lowerBoundParameter.setValue(Double.NEGATIVE_INFINITY);
+      }
+      lowerBoundParameter.setExplicitlyConstant(true);
+      lowerBoundParameter.setUnits(CobraConstants.mmol_per_gDW_per_hr);
+    }
+    
+    LocalParameter objectiveCoefficientParameter = kineticLaw.getLocalParameter(CobraConstants.OBJECTIVE_COEFFICIENT);
+    if (objectiveCoefficientParameter == null)
+    {
+      objectiveCoefficientParameter = kineticLaw.createLocalParameter(CobraConstants.OBJECTIVE_COEFFICIENT);
+      objectiveCoefficientParameter.setUnits(CobraConstants.mmol_per_gDW_per_hr);
+      objectiveCoefficientParameter.setValue(0d);
+    }
+  }
+    
+  private void unsetConstantForSpeciesReferencesOfReaction(Reaction reaction) {
+    // unset attribute constant for products and reactants of reaction
+    for (SpeciesReference speciesReference: reaction.getListOfProducts()) {
+      speciesReference.unsetConstant();
+    }
+    for (SpeciesReference speciesReference: reaction.getListOfReactants()) {
+      speciesReference.unsetConstant();
+    }
+  }
+   
+  private void updateKineticLawFromBound (Reaction reaction, FluxBound bound)
+  {
+    if (reaction == null || bound == null) {
+      return;
+    }
+    Operation operation = bound.getOperation();
+    KineticLaw kineticLaw = reaction.getKineticLaw();
+    LocalParameter lowerBoundParameter = kineticLaw.getLocalParameter(CobraConstants.LOWER_BOUND);
+    LocalParameter upperBoundParameter = kineticLaw.getLocalParameter(CobraConstants.UPPER_BOUND);
+    
+    if (operation.equals(FluxBound.Operation.LESS_EQUAL) || operation.equals(FluxBound.Operation.EQUAL)) 
+    {
+      upperBoundParameter.setValue(bound.getValue());
+    }
+    
+    if (operation.equals(FluxBound.Operation.GREATER_EQUAL) || operation.equals(FluxBound.Operation.EQUAL)) 
+    {
+      lowerBoundParameter.setValue(bound.getValue());
+    }
```

Comment:
in jsbml only these three operations are defined (see org.sbml.jsbml.ext.fbc.FluxBound). They are now all covered. libSBML also checks for "less" and "greater" upon conversion, but these do not seem to be present in jsbml. The converter from FBCv2 to FBCv1 in jsbml only creates LESS_EQUAL and GREATER_EQUAL operations.